src/fragmentarium/infrastructure/FragmentRepository.ts

```diff
@@ -90,6 +97,32 @@ class ApiFragmentRepository
     return this._fetch({ number })
   }
 
+  searchReference(id: string, pages: string): FragmentInfosPromise {
+    return this._fetch({ id, pages }).then((fragInfos) => {
+      const newFragInfo: any = produce(
```

Comment:
Do not use `any`. Is this supposed to be `FragmentInfo` or `FragmentInfo[]`? The whole method looks rather convoluted and could be refactored to be more readable.